doc/doc-site/docs/off-chain/dss/gcp.md

```diff
@@ -0,0 +1,916 @@
+   gcloud config set artifacts/location $REGION
+   gcloud config set compute/zone $ZONE
+   gcloud config set run/region $REGION
+   ```
+
+### Build images
+
+1. Create a [GCP Artifact Registry](https://cloud.google.com/artifact-registry/docs/overview) Docker repository named `images`:
+
+   ```sh
+   gcloud artifacts repositories create images \
+       --repository-format docker
+   ```
+
+1. Set the repository as default:
+
+   ```sh
+   gcloud config set artifacts/repository images
+   ```
+
+1. Clone the Econia repository:
+
+   ```sh
+   git clone https://github.com/econia-labs/econia.git
+   ```
+
+1. Build the DSS images from source:
+
+   ```sh
+   gcloud builds submit econia \
+       --config econia/src/docker/gcp-tutorial-config.yaml \
+       --substitutions _REGION=$REGION
+   ```
+
+   :::tip
+   This will take a while, since it involves the compilation of several binaries from source.
+   :::
+
+### Create bootstrapper
+
+1. Create a [GCP Compute Engine instance](https://cloud.google.com/compute/docs/instances) for bootstrapping config files, with two attached [persistent disks](https://cloud.google.com/compute/docs/disks):
+
+   ```sh
+   gcloud compute instances create bootstrapper \
+       --create-disk "$(printf '%s' \
+            auto-delete=no,\
+            name=postgres-disk,\
+            size=100GB\
+       )" \
+       --create-disk "$(printf '%s' \
+            auto-delete=no,\
+            name=processor-disk,\
+            size=1GB\
+       )"
+   ```
+
+1. [Create an SSH key pair](https://cloud.google.com/compute/docs/connect/create-ssh-keys) and use it to upload PostgreSQL configuration files to the bootstrapper:
+
+   ```sh
+   mkdir ssh
+   ssh-keygen -t rsa -f ssh/gcp -C bootstrapper -b 2048 -q -N ""
+   gcloud compute scp \
+       econia/src/docker/database/configs/pg_hba.conf \
+       econia/src/docker/database/configs/postgresql.conf \
+       bootstrapper:~ \
+       --ssh-key-file ssh/gcp
+   ```
+
+1. [Connect to](https://cloud.google.com/compute/docs/connect/standard-ssh) the bootstrapper instance:
+
+   ```sh
+   gcloud compute ssh bootstrapper --ssh-key-file ssh/gcp
+   ```
+
+1. [Check connected disks](https://cloud.google.com/compute/docs/disks/format-mount-disk-linux#format_linux):
+
+   ```sh
+   sudo lsblk
+   ```
+
+   :::tip
+   The device name for the `postgres` disk will probably be `sbd`, and the device name for the `processor` will probably be `sdc` (check the disk sizes if you are unsure).
+   :::
+
+1. Store the device names in shell variables:
+
+   ```sh
+   POSTGRES_DISK_DEVICE_NAME=<PROBABLY_sdb>
+   PROCESSOR_DISK_DEVICE_NAME=<PROBABLY_sdc>
+   ```
+
+   ```sh
+   echo "PostgreSQL disk device name: $POSTGRES_DISK_DEVICE_NAME"
+   echo "Processor disk device name: $PROCESSOR_DISK_DEVICE_NAME"
+   ```
+
+1. [Format and mount the disks with read/write permissions](https://cloud.google.com/compute/docs/disks/format-mount-disk-linux#format_linux):
+
+   ```sh
+   sudo mkfs.ext4 \
```

Comment:
I would space these all out by one line each.

Reply:
Addressed in f65f664